### ISSUE TITLE
Leave-One-Out Cross-Validation機能を追加

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidata#30]
### 概要

Leave-One-Out Cross-Validation（LOOCV）による訓練・評価機能を `ClassificationModeler` に追加する。

### 背景

- 現在はホールドアウト法とK分割交差検証のみ対応
- サンプル数が少ないデータセットでは、LOOCVによる評価がより適切な場合がある
- 各サンプルに対する予測結果を取得したいユースケースがある（PredictionExporterとの連携）

#### train_model() への LOOCV オプション追加

```python
modeler.train_model(
    data=df,
    feature_names=feature_names,
    target_column="target",
    validation_method="loocv",  # 新規オプション
)
```

#### 出力内容

- 各サンプルに対する予測結果（true_label, predicted_label, correct_flag）
- 全体の精度（Accuracy）
- クラスごとの評価指標（Precision, Recall, F1）

#### predictions_loocv.csv の自動生成

`save_model_results()` 実行時に以下の形式で出力:

| 元データの列... | true_label | predicted_label | correct_flag |
|----------------|------------|-----------------|--------------|
| ...            | setosa     | setosa          | 1            |
| ...            | versicolor | virginica       | 0            |

### 技術的な考慮事項

- `sklearn.model_selection.LeaveOneOut` を使用
- サンプル数が多い場合は計算コストが高いため、警告を表示（例: n > 1000）

### 対象ファイル

| ファイル | 変更内容 |
|---------|---------|
| `pochidata/core/classification_modeler.py` | LOOCV訓練ロジック追加 |
| `pochidata/core/prediction_exporter.py` | LOOCV用出力メソッド追加（必要に応じて） |
| `tests/core/test_classification_modeler.py` | LOOCVテスト追加 |

### 完了条件

- [ ] `validation_method="loocv"` でLOOCV訓練が実行できる
- [ ] 各サンプルの予測結果が取得できる
- [ ] `predictions_loocv.csv` が自動生成される
- [ ] サンプル数が多い場合に警告が表示される
- [ ] ユニットテストが追加され、パスしている
- [ ] 既存テストがパスしている

### ブランチ

`feature/loocv`
